### PR TITLE
OTG-4.8.14.1 - v4 migration cleanup

### PR DESCRIPTION
**Pull request by rejahrehim**
This PR part of issue #84 

- [ ] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Fix links
- Fix linting

### CODE REVIEW

File: document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.14.1_Testing_for_Heap_Overflow.md [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
 
 When a heap-based buffer is overflowed the control information in these tags is overwritten. When the heap management routine frees the buffer, a memory address overwrite takes place leading to an access violation. When the overflow is executed in a controlled fashion, the vulnerability would allow an adversary to overwrite a desired memory location with a user-controlled value. In practice, an attacker would be able to overwrite function pointers and various addresses stored in structures like GOT, .dtors or TEB with the address of a malicious payload.
 
 There are numerous variants of the heap overflow (heap corruption) vulnerability that can allow anything from overwriting function pointers to exploiting memory management structures for arbitrary code execution. Locating heap overflows requires closer examination in comparison to stack overflows, since there are certain conditions that need to exist in the code for these vulnerabilities to be exploitable.
 
-How to Test
------------
+## How to Test
 
 ### Black Box testing
```
Comment (kingthorin) [inline document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.14.1_Testing_for_Heap_Overflow.md]: Title caps please

Comment (kingthorin) [reply document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.14.1_Testing_for_Heap_Overflow.md]: On the Grey Box heading too

Comment (rejahrehim) [reply document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.14.1_Testing_for_Heap_Overflow.md]: @kingthorin Fixed

File: document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.14.1_Testing_for_Heap_Overflow.md [hunk 2]
```diff
@@ -1,85 +1,87 @@
...
 When reviewing code, one must realize that there are several avenues where heap related vulnerabilities may arise. Code that seems innocuous at the first glance can actually be vulnerable under certain conditions. Since there are several variants of this vulnerability, we will cover only the issues that are predominant.
 
-Most of the time, heap buffers are considered safe by a lot of developers who do not hesitate to perform insecure operations like strcpy( ) on them. The myth that a stack overflow and instruction pointer overwrite are the only means to execute arbitrary code proves to be hazardous in case of code shown below:-
+Most of the time, heap buffers are considered safe by a lot of developers who do not hesitate to perform insecure operations like `strcpy( )` on them. The myth that a stack overflow and instruction pointer overwrite are the only means to execute arbitrary code proves to be hazardous in case of code shown below:-
 
-    int main(int argc, char *argv[])
-        {
-            ……
+```c
+int main(int argc, char *argv[])
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.14.1_Testing_for_Heap_Overflow.md]: Can you please update to this?
```c
int main(int argc, char *argv[])
    {
        vulnerable(argv[1]);
        return 0;
    }


    int vulnerable(char *buf)
    {
        HANDLE hp = HeapCreate(0, 0, 0);
        HLOCAL chunk = HeapAlloc(hp, 0, 260); 
        strcpy(chunk, buf);  // Vulnerability here!
        return 0;
    }
```

File: document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.14.1_Testing_for_Heap_Overflow.md [hunk 3]
```diff
@@ -1,85 +1,87 @@
...
 
-The malloc in line 1 allocates memory based on the value of length, which happens to be a 32 bit integer. In this particular example, length is user-controllable and a malicious TNEF file can be crafted to set length to ‘-1’, which would result in malloc( 0 ). Therefore, this malloc would allocate a small heap buffer, which would be 16 bytes on most 32 bit platforms (as indicated in malloc.h).
+Lately, several products, especially anti-virus libraries, have been affected by variants that are combinations of an integer overflow and copy operations to a heap buffer. As an example, consider a vulnerable code snippet, a part of code responsible for processing `TNEF` filetypes, from Clam Anti Virus 0.86.1, source file `tnef.c` and function `tnef_message( )`:
 
-And now, in line 2, a heap overflow occurs in the call to fread( ). The 3rd argument, in this case length, is expected to be a size\_t variable. But if it’s going to be ‘-1’, the argument wraps to 0xFFFFFFFF, thus copying 0xFFFFFFFF bytes into the 16 byte buffer.
+```c
+string = cli_malloc(length + 1); ''' Vulnerability'''
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.8_Input_Validation_Testing/4.8.14.1_Testing_for_Heap_Overflow.md]:
```c
string = cli_malloc(length + 1); // Vulnerable
if(fread(string, 1, length, fp) != length) { // Vulnerable
    free(string);
    return -1;
}

**Pull request merged**